File: week1/assignments/to-lower-case/soln.cpp
#include <string>
#include <locale>
#include <iostream>

using namespace std;

class Solution {
    public:
        string toLowerCase(string s) {
            locale loc;
            string test = "";
            for (string::size_type i = 0; i < s.length(); i++)
                test += tolower(s[i], loc);
            return test;
        }
};

int main(int argc, char* argv[]) {
    Solution soln;

    cout << soln.toLowerCase("aBcDeFgHiJ") << endl;
    cout << soln.toLowerCase("abcdefg") << endl;
    cout << soln.toLowerCase("ABCDEFG") << endl;

    return 0;
}

File: week1/assignments/reverse-integer/soln.py
class Solution:
    def reverse(self, x: int) -> int:
        return int( str( abs(x) )[::-1] )



if __name__ == "__main__":
    soln = Solution()

    test_cases = [
        123,
        -123,
        120,
        1200,
        -320
    ]

    for test in test_cases:
        print(f"{test} -> {soln.reverse(test)}")